PhysicsTools/Heppy/python/analyzers/objects/METAnalyzer.py

```diff
@@ -66,9 +66,37 @@ def makeTkMETs(self, event):
         chargedchs = []
         chargedPVLoose = []
         chargedPVTight = []
-        dochs=getattr(self.cfg_ana,"includeTkMetCHS",True)       
-        dotight=getattr(self.cfg_ana,"includeTkMetPVTight",True)       
-        doloose=getattr(self.cfg_ana,"includeTkMetPVLoose",True)       
+        chargedNoPV = []
+        chargedPVUsedInFit = []
+        chargedUsedInFitTight = []
+        chargedUsedInFitLoose = []
+        chargedCompatibilityDz = []
+        chargedCompatibilityBTag = []
+        chargedNotReconstructedPrimary = []
+        chargedOtherDeltaZ = []
+        dochs=getattr(self.cfg_ana,"includeTkMetCHS",True)
+        dotight=getattr(self.cfg_ana,"includeTkMetPVTight",True)
+        doloose=getattr(self.cfg_ana,"includeTkMetPVLoose",True)
+        doPVUsedInFit=getattr(self.cfg_ana,"includeTkMetPVUsedInFit",True)
+        doNoPV=getattr(self.cfg_ana,"includeTkMetNoPV",True)
+        doUsedInFitTight=getattr(self.cfg_ana,"includeTkMetUsedInFitTight",True)
+        doUsedInFitLoose=getattr(self.cfg_ana,"includeTkMetUsedInFitLoose",True)
+        doCompatibilityDz=getattr(self.cfg_ana,"includeTkMetCompatibilityDz",True)
+        doCompatibilityBTag=getattr(self.cfg_ana,"includeTkMetCompatibilityBTag",True)
+        doNotReconstructedPrimary=getattr(self.cfg_ana,"includeTkMetNotReconstructedPrimary",True)
+        doOtherDeltaZ=getattr(self.cfg_ana,"includeTkMetOtherDeltaZ",True)
+        doPuppichs=getattr(self.cfg_ana,"includeTkMetPuppiCHS",True)
+        doPuppitight=getattr(self.cfg_ana,"includeTkMetPuppiPVTight",True)
+        doPuppiloose=getattr(self.cfg_ana,"includeTkMetPuppiPVLoose",True)
+        doPuppiPVUsedInFit=getattr(self.cfg_ana,"includeTkMetPuppiPVUsedInFit",True)
+        doPuppiNoPV=getattr(self.cfg_ana,"includeTkMetPuppiNoPV",True)
+        doPuppiUsedInFitTight=getattr(self.cfg_ana,"includeTkMetPuppiUsedInFitTight",True)
+        doPuppiUsedInFitLoose=getattr(self.cfg_ana,"includeTkMetPuppiUsedInFitLoose",True)
+        doPuppiCompatibilityDz=getattr(self.cfg_ana,"includeTkMetPuppiCompatibilityDz",True)
+        doPuppiCompatibilityBTag=getattr(self.cfg_ana,"includeTkMetPuppiCompatibilityBTag",True)
+        doPuppiNotReconstructedPrimary=getattr(self.cfg_ana,"includeTkMetPuppiNotReconstructedPrimary",True)
+        doPuppiOtherDeltaZ=getattr(self.cfg_ana,"includeTkMetPuppiOtherDeltaZ",True)
+
```

Comment:
Some of these are also in the default config.
Also, the default config is false, but the default in the code is true.
I would leave the default to false also in the code, as I think they're mostly for MET commissioning, the average user won't want to run all them, right?